csp/middleware.py

```diff
@@ -23,6 +26,14 @@ class CSPMiddleware(MiddlewareMixin):
     See http://www.w3.org/TR/CSP/
 
     """
+    def _make_nonce(self, request, length=16):
+        # Return cached value if this has been executed already
+        if not getattr(request, '_csp_nonce', None):
+            request._csp_nonce = get_random_string(length)
```

Comment:
Why do we need to cache this? afaik django processes every middleware exactly once so saving this might be unnecessary.